Src/CSharpier.Tests/FormattingTests/TestFiles/cs/VariableDeclarations.test

```diff
@@ -120,8 +120,8 @@ class ClassName
                     : null
         ) { }
 
-        var variableDeclarator = conditionalAccessExpression?.invocationExpression(
-            o => someLongValue___________________________________
+        var variableDeclarator = conditionalAccessExpression?.invocationExpression(o =>
+            someLongValue___________________________________
```

Comment:
So, I think in the case where the parameter declaration and the expression body both fit in the one line, it's probably better to leave it there, than forcefully separate them by moving parameter declaration to the line above.

Reply:
It is tricky to implement so that we save the following formatting:
```C#

var y = someList.Where(o =>
    someLongValue_______________________
    && theseShouldNotIndent_________________
    && theseShouldNotIndent_________________
        > butThisOneShould_________________________________________
);

b.HasKey(l => new
{
    l.UserId,
    l.LoginProvider,
    l.Name
});
```

Also, this synthetic sample looks better, if it's formatted like that:
```C#
var variableDeclarator = conditionalAccessExpression
    ?.invocationExpression(o => 
        someLongValue___________________________________
    );
```

I've tried to find such case in my code base and, in my opinion, separating of the parameter and the body looks better

#### Example 1
Here if we separate the parameter and the body, the parameter is good aligned with the first lambda and the distance between the parameter and its first usage is still small.
And if we move them both to the next line, we just get longer line.
```C#
    var patient = await _context.Patients
        .Include(x => x.Questionnaires)
        .Include(x => 
            x.Statistics.Where(s => s.TherapyDay >= dateSelector).OrderBy(s => s.TherapyDay)
        );

    // vs

    var patient = await _context.Patients
        .Include(x => x.Questionnaires)
        .Include(
            x => x.Statistics.Where(s => s.TherapyDay >= dateSelector).OrderBy(s => s.TherapyDay)
        );

```

#### Example 2
If the lambda doesn't fit in the one line, must the parameter jump back or take its own line?

If it must jump back then it jumps back and forward, while the body is growing, and it can look strange.

If the parameter must take its own line, then we get an unnecessary indent and an almost blank line
```C#
    var patient = await _context.Patients
        .Include(x =>
            x.Statistics.Where(s => s.TherapyDay >= dateSelector).OrderBy(s => s.TherapyDay)
        )
        .Include(x => 
            x.Questionnaires.Where(q => 
                q.Survey.Type == SurveyType.DailySurvey.ToSurveyTypeString()
            )
        );

    // vs

    var patient = await _context.Patients
        .Include(
            x => x.Statistics.Where(s => s.TherapyDay >= dateSelector).OrderBy(s => s.TherapyDay)
        )
        .Include(x => 
            x.Questionnaires.Where(q => 
                q.Survey.Type == SurveyType.DailySurvey.ToSurveyTypeString()
            )
        );

    // vs 

    var patient = await _context.Patients
        .Include(
            x => x.Statistics.Where(s => s.TherapyDay >= dateSelector).OrderBy(s => s.TherapyDay)
        )
        .Include(
            x => 
                x.Questionnaires.Where(
                    q => q.Survey.Type == SurveyType.DailySurvey.ToSurveyTypeString()
                )
        );
```

Reply:
Alright, fair enough. You've put a lot more thought into this than I have :shrug: 
Assuming @belav has no objections, I'll merge this PR tonight.

Reply:
I haven't had a chance to look yet, but it is still on my todo list!